backend/src/services/profile_service.py

```
@@ -0,0 +1,207 @@
+        log.info("Creating profile")
+
+        self._validate_experience_years(payload.experience_years)
+        self._validate_skills(payload.skills)
+
+        try:
+            existing = await self.repo.get_first()
+            if existing is not None:
+                log.bind(profile_id=existing.id).warning("Profile already exists")
+                raise BusinessLogicError(
+                    "Profile already exists. Only one profile is allowed."
+                )
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)